### PR TITLE
Mes 6231 load app info

### PR DESCRIPTION
**Pull request by piercemorris**
## Description

Lift and shift app-info state management content where necessary, hand copied functions to use the new NGRX syntax. Updated karma to show test descriptions rather than the generic information. Update tests to pass for the store module.

NB all functions have been shifted over but not currently being used until needed, only 'load app version' action/effect is being used once the platform is loaded in the app.

## Checklist

- [x] PR title includes the JIRA ticket number
- [x] Branch is rebased against the latest develop
- [x] Code has been tested manually
- [x] PR link added to JIRA ticket
- [ ] One review from each scrum team
- [ ] Squashed commit contains the JIRA ticket number

### CODE REVIEW

File: src/app/providers/app-info/__mocks__/app-version.mock.ts [hunk 1]
```diff
@@ -0,0 +1,9 @@
+// eslint-disable-next-line import/no-cycle
+import { APP_VERSION_NUMBER } from '../__tests__/app-info.spec';
```
Comment (szabikr) [inline src/app/providers/app-info/__mocks__/app-version.mock.ts]: You could extract this `APP_VERSION_NUMBER` from the spec file into a mock file or into constants

File: src/app/providers/app-info/__tests__/app-info.spec.ts [hunk 2]
```diff
@@ -0,0 +1,48 @@
+import { TestBed } from '@angular/core/testing';
+import { configureTestSuite } from 'ng-bullet';
+import { AppVersion } from '@ionic-native/app-version/ngx';
+import { AppInfoProvider } from '../app-info';
+// eslint-disable-next-line import/no-cycle
```
Comment (szabikr) [inline src/app/providers/app-info/__tests__/app-info.spec.ts]: This line is no longer needed if you extract `APP_VERSION_NUMBER` from the spec file

File: src/app/providers/app-info/__tests__/app-info.spec.ts [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+    TestBed.configureTestingModule({
+      providers: [
+        AppInfoProvider,
+        { provide: AppVersion, useClass: AppVersionMock },
+      ],
+    });
+  });
+
+  beforeEach(() => {
+    appInfoProvider = TestBed.get(AppInfoProvider);
```
Comment (szabikr) [inline src/app/providers/app-info/__tests__/app-info.spec.ts]: Isn't `TestBed.get` deprecated in our Angular version?

File: src/store/app-info/app-info.reducer.ts [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+
+export const initialState: AppInfoStateModel = {
+  versionNumber: 'VERSION_NOT_LOADED',
+  employeeId: null,
+  employeeName: 'Unknown Name',
+};
+
+export const appInfoReducer = createReducer(
+  initialState,
+  on(LoadAppInfoSuccess, (state: AppInfoStateModel, { versionNumber }) => ({
```
Comment (szabikr) [inline src/store/app-info/app-info.reducer.ts]: Using the new reducer syntax, great!

**Review by szabikr — COMMENTED** [review]
Great work!

Once the `APP_VERSION_NUMBER` is extracted from the spec file I'm happy to approve.

**Review by tomcrawleyy — APPROVED** [review]

File: src/app/app.component.ts [hunk 5]
```diff
@@ -20,6 +25,8 @@ export class AppComponent {
 
   initializeApp() {
     this.platform.ready().then(() => {
+      this.store$.dispatch(LoadAppVersion());
+      this.store$.dispatch(LoadEmployeeName());
```
Comment (szabikr) [inline src/app/app.component.ts]: It's a bit too early for dispatching `LoadEmployeeName` as the user is not logged in just yet. Will always be unknown at this point.

A good place to dispatch this action is after the user is authenticated and the token is acquired.

**Review by szabikr — APPROVED** [review]
Great work!

**Pull request merged**